### ISSUE TITLE
Abstract base schemas for use exclusively with discriminators

### ISSUE THREAD

**Issue by zbjornson** [Automattic/mongoose#5465]
**Do you want to request a *feature* or report a *bug*?**

*feature*

It would be nice to have abstract schemas that you cannot instantiate, and instead must instantiate via discriminators.

```js
var shapeSchema = new Schema({
  name: String
}, {
  discriminatorKey: "type",
  abstract: true // <----
});

var Shape = db.model('Shape', shapeSchema);
var Circle = Shape.discriminator('Circle', new Schema({ radius: Number }));
```

`new Shape()` would be *invalid*
`Shape.create({type: "not a registered discriminator"})` would be *invalid*
`Shape.create({type: "Circle"})` would be *valid*
`new Circle()` would be *valid*
`Shape.find()` would be *valid*

If you agree that this functionality would be useful, I can try to work on a PR.

---

Current workaround is to have a helper function used before `Shape.create` that makes sure the value for the discriminator key is allowed.

**Comment by vkarpov15** [comment]
I like the general idea, but how would `abstract` work if you set it on a schema you pass to `discriminator()`?

**Comment by zbjornson** [comment]
> how would abstract work if you set it on a schema you pass to discriminator()

I think that should throw an error. (Can you think of a use case?) Unless someone is creating schemas at runtime, that would be a boot-time error that would be easy to spot and remedy.

**Comment by vkarpov15** [comment]
Yeah that's a fair point, throwing an error seems sane. Can you implement this as a plugin? http://thecodebarbarian.com/2015/03/06/guide-to-mongoose-plugins Should be pretty easy to just throw it in the `lib/plugins` folder

**Comment by Faibk** [comment]
@zbjornson Did you ever manage to find time to implement this as a plugin?

**Comment by zbjornson** [comment]
@Faibk I haven't had a chance to yet, no. We use a static `BaseModel.create` method currently to achieve this effect for now.

**Comment by Itee** [comment]
Something new about abstract schema ? I have exactly the same need than @zbjornson.
